nemo/collections/asr/parts/k2/loss_mixins.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from typing import List, Optional, Tuple
+
+import torch
+
+from nemo.collections.asr.parts.k2.grad_utils import GradExpNormalize
+from nemo.collections.asr.parts.k2.utils import (
+    create_supervision,
+    get_arc_weights,
+    get_uniform_rnnt_prune_ranges,
+    make_non_pad_mask,
+    make_non_pad_mask_3d,
+    prep_padded_densefsavec,
+)
+
+# use k2 import guard
+# fmt: off
+from nemo.core.utils.k2_utils import k2_import_guard # isort:skip
+k2_import_guard()
+import k2 # isort:skip
+# fmt: on
+
+
+class CtcK2Mixin(ABC):
```

Comment:
As we discussed: it seems to be better to have a common (abstract) interface for mixins, since `MLLoss` and `*K2Mixin` are partial implementations. (I write it here just not to forget to review it further)

Reply:
Let's return to this issue in some following PRs.